packages/@lwc/template-compiler/src/__tests__/parser.spec.ts

```diff
@@ -465,6 +465,34 @@ describe('expression', () => {
         });
     });
 
+    it('quoted expression with leading spaces should produce error', () => {
+        const { warnings } = parseTemplate(`<template><input title="  {myValue}" /></template>`);
+        expect(warnings[0].message).toMatch(`Ambiguous attribute value title="  {myValue}"`);
+        expect(warnings[0]).toMatchObject({
+            location: EXPECTED_LOCATION,
+        });
+    });
+    // if the curly brace is at 0th or 1st position of the string, then it is currenlty an error (given closing brace exists)
+    // "{username}-lightningUpload-{recId}-"
+    // "[{asdf}]"
+    it('quoted expression with leading characters should not produce error', () => {
```

Comment:
The question is not only should it throw an error, but also when it doesn't throw an error, should it produce a string or an expression attribute value.

```suggestion
    it('quoted expression with leading characters should a string attribute value', () => {
```